lib/Modules/ADHooks/ADHooks.psm1

```diff
@@ -1299,31 +1296,70 @@ function Set-NodesKCD {
         Write-JujuWarning "Remote charm didn't set any constraints delegations to be set by AD"
         return
     }
-
     if(!$computerNames) {
         Write-JujuWarning "No computers names set for the relation: $RelationId"
         return
     }
 
     $constraints = Get-UnmarshaledObject $marshaledConstraints
+    $nodesKCDMarshaled = Get-LeaderData -Attribute "nodes-kcd-$RelationId"
```

Comment:
this has the potential to grow quite a lot. I would recommend we use individual keys for each host. I am not sure of the limits of the value for a relation key in juju.